### PR TITLE
PLFM-5617: Send reset password email to the alias address

### PR DESCRIPTION
**Pull request by marcomarasca**
- Added the possibility to specify a PrincipalAlias in the MessageManager
- Added relevant unit and integration tests
- Fixed existing related integration tests to actually check the result

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManager.java [hunk 1]
```diff
@@ -102,11 +103,30 @@ public MessageToUser forwardMessage(UserInfo userInfo, String messageId,
...
-	 * Sends a password reset email based on a template via Amazon SES
+	 * Sends a password reset email based on a template via Amazon SES, using as destination the default
+	 * notification email of the user identified by the given reset token
+	 * 
+	 * @param passwordResetPrefix The url prefix for the verification callback in the portal
+	 * @param passwordResetToken  The reset token generated for the user that needs the password reset
 	 */
-	public void sendNewPasswordResetEmail(String passwordResetPrefix, PasswordResetSignedToken passwordResetToken) throws NotFoundException;
+	public void sendNewPasswordResetEmail(String passwordResetPrefix, PasswordResetSignedToken passwordResetToken);
+
```
Comment (marcomarasca) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManager.java]: Dead code: can be removed

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java [hunk 2]
```diff
@@ -714,9 +722,51 @@ public void sendDeliveryFailureEmail(String messageId, List<String> errors) thro
...
 	}
 	
-	
+	private String getEmailForAlias(PrincipalAlias alias) throws NotFoundException {
+		if (AliasType.USER_EMAIL.equals(alias.getType())) {
+			return alias.getAlias();
+		}
+		return getEmailForUser(alias.getPrincipalId());
+	}
+
```
Comment (marcomarasca) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java]: Add test for this

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java [hunk 3]
```diff
@@ -714,9 +722,51 @@ public void sendDeliveryFailureEmail(String messageId, List<String> errors) thro
...
+		sesClient.sendRawEmail(sendEmailRequest);
+	}
+
+	private static String getPasswordResetUrl(String passwordResetUrlPrefix,
+			PasswordResetSignedToken passwordResetToken) {
+		String resetUrl = passwordResetUrlPrefix + SerializationUtils.serializeAndHexEncode(passwordResetToken);
+		EmailUtils.validateSynapsePortalHost(resetUrl);
+		return resetUrl;
+	}
```
Comment (marcomarasca) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java:750]: Add test for this method

**Comment by marcomarasca** [comment]
@john-hill: Added fix commits from the code review!

**Review by john-hill — APPROVED** [review]
Looks good!

**Pull request merged**